pkgs/applications/networking/ouroboros/default.nix

```diff
@@ -0,0 +1,44 @@
+{ stdenv
+, lib
+
+, cmake
+, dnsutils
+, fetchgit
+, fuse
+, libgcrypt
+, pkgconfig
+, protobuf
+, protobufc
+, systemd
+}:
+stdenv.mkDerivation rec {
+  pname = "ouroboros";
+  version = "0.19.3";
+  src = fetchgit {
+    url = "git://ouroboros.rocks/${pname}";
+    rev = version;
+    hash = "sha256-DYkQHmfAOxoCA6AqIunHgx7SbeRxpwlPT7ztRzLgv10=";
+  };
+  buildInputs = [
+    cmake
+    dnsutils
+    fuse
+    libgcrypt
+    pkgconfig
+    protobuf
+    protobufc
+    systemd
+  ];
+  cmakeFlags = [ "-DSYSTEMD_UNITDIR=${placeholder "out"}/etc/systemd/system" ];
```

Comment:
This is strange, is this going to try to write to ``/etc/systemd/system`` ?

Reply:
Yes, the install step tries to install systemd unit files for the irmd service into `$(SYSTEMD_UNITDIR)`.  I'm not entirely sure this is the best path for them; do you have better suggestions?  The nixpkgs systemd setup seems to pick up unit files from here.

Reply:
Nix build derivations in a sandbox, where you cannot access `/etc`.

Reply:
Looking at my system, other packages seem to be divided on the matter:
- some install to `lib/systemd/system`
  - mdmonitor, mdadm
  - nix-daemon
- some install to `etc/systemd/system`
  - dbus
  - polkit
  - rtkit
  - accounts-daemon
- systemd itself has a bunch of services in `example/systemd/system`
  - I assume this is a special case, probably not the one to go for
- others have a dedicated output that is a directory containing just the unit
  - I don't really understand why, vs having the output be the service directly — maybe to avoid future breakage of dependents if they need to add more files (`nix-support` et cetera?)

Reply:
Hey @Twey, the way to setup systemd units on nixos is to create a nixos module ! You can't write to `/etc` from inside a nix build because they are performed in a sandbox.

Reply:
Sorry, I think I misapprehended your confusion.  This line does not write to `/etc` (the global path), which would not work of course :slightly_smiling_face:, but `${placeholder "out"}/etc`, a local path under the `out` output for this derivation.  In doing so it follows a pattern adhered to by several — but not all — other packages in `nixpkgs`.  There are some alternative conventions I've found in `nixpkgs` that I have listed above; I don't have a particular preference, so if you prefer one or the other I have no real qualms about changing this package to adhere to a different one.

But either way I don't think anybody is suggesting this package should install to the global `/etc` :slightly_smiling_face:

Reply:
We certainly could make a NixOS module for Ouroboros, but it's not done in this PR.  If we did, I'd strongly suggest we should use the systemd unit produced by the existing Ouroboros build process (and placed in `${SYSTEMD_UNITDIR}`), rather than reverse-engineering it and duplicating the logic into the module.